src/node/util.ts

```diff
@@ -4,8 +4,182 @@
 
 import { default as internalTypes } from 'node-internal:internal_types';
 
+import {
+  validateFunction
+} from 'node-internal:validators';
+
+import {
+  ERR_FALSY_VALUE_REJECTION,
+  ERR_INVALID_ARG_TYPE,
+} from 'node-internal:internal_errors';
+
+export {
+  format
+} from 'node-internal:internal_format';
+
+import { format } from 'node-internal:internal_format';
+
 export const types = internalTypes;
+;
+const callbackifyOnRejected = (reason: unknown, cb : Function) => {
+  if (!reason) {
+    reason = new ERR_FALSY_VALUE_REJECTION(`${reason}`);
+  }
+  return cb(reason);
+};
+
+export function callbackify
+    <T extends (...args: any[]) => Promise<any>>(original: T):
+    T extends (...args: infer TArgs) => Promise<infer TReturn> ? (...params: [...TArgs, (err: Error, ret: TReturn) => any]) => void : never {
+  validateFunction(original, 'original');
+
+  function callbackified(this: unknown,
+                         ...args: [...unknown[],
+                         (err: unknown, ret: unknown) => void]) : any {
+    const maybeCb = args.pop();
+    validateFunction(maybeCb, 'last argument');
+    const cb = (maybeCb as Function).bind(this);
+    Reflect.apply(original, this, args)
+      .then((ret: any) => queueMicrotask(() => cb(null, ret)),
+            (rej: any) => queueMicrotask(() => callbackifyOnRejected(rej, cb)));
+  }
+
+  const descriptors = Object.getOwnPropertyDescriptors(original);
+  if (typeof descriptors['length']!.value === 'number') {
+    descriptors['length']!.value++;
+  }
+  if (typeof descriptors['name']!.value === 'string') {
+    descriptors['name']!.value += 'Callbackified';
+  }
+  const propertiesValues = Object.values(descriptors);
+  for (let i = 0; i < propertiesValues.length; i++) {
+    Object.setPrototypeOf(propertiesValues[i], null);
+  }
+  Object.defineProperties(callbackified, descriptors);
+  return callbackified as any;
+}
+
+const kCustomPromisifiedSymbol = Symbol.for('nodejs.util.promisify.custom');
+const kCustomPromisifyArgsSymbol = Symbol('customPromisifyArgs');
+
+// TODO(later): Proper type signature for promisify.
+export function promisify(original: Function): Function {
```

Comment:
I wanted to suggest to copy the one from definitelytyped, but it is huge:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/util.d.ts#L1050

I'm thinking maybe for node-compat we need to install `@types/node` and use those?

Reply:
Yeah, I had looked there also and quickly ran away.